pkg/model/trace/matches.go

```diff
@@ -114,12 +99,37 @@ func MatchesProto(id []byte, trace *tempopb.Trace, req *tempopb.SearchRequest) (
 	}, nil
 }
 
-// searchAttributes returns an int that indicates if the passed tags match the trace attributes
-//  possible values: attributesNotFound, attributesFailedMatch, attributesPassedMatch
-func searchAttributes(tags map[string]string, atts []*v1common.KeyValue) int {
-	// start with the assumption that we won't find any matching attributes
-	allMatch := attributesNotFound
+func allTagsFound(tagsToFind map[string]string) bool {
+	return len(tagsToFind) == 0
+}
+
+// matchSpan searches for reserved tag names and maps them to actual
+// properties of the span. it removes any matches it finds from
+// the provided tags map
+func matchSpan(tags map[string]string, s *v1.Span) {
+	if name, ok := tags[search.SpanNameTag]; ok {
+		if name == s.Name {
```

Comment:
I think this should be case-insensitive substring match, if we want to match ingester search exactly.  Don't feel strongly, as this is more accurate and seems fine.  Same note applies to `matchAttributes`, case-insensitivity.